syllabus/examples/experimental/sb3_procgen_plr.py

```diff
@@ -47,70 +213,149 @@ def __init__(self, curriculum, verbose=0):
+    )
 
+    # TRY NOT TO MODIFY: seeding
+    random.seed(args.seed)
+    np.random.seed(args.seed)
+    torch.manual_seed(args.seed)
+    torch.backends.cudnn.deterministic = args.torch_deterministic
 
-sample_env = gym.make("procgen-bigfish-v0")
-sample_env = ProcgenTaskWrapper(sample_env)
-curriculum = CentralizedPrioritizedLevelReplay(sample_env.task_space, num_processes=64, num_steps=256)
-curriculum, task_queue, update_queue = make_multiprocessing_curriculum(curriculum)
-venv = DummyVecEnv(
+    device = torch.device("cuda" if torch.cuda.is_available() and args.cuda else "cpu")
+    print("Device:", device)
+
+    # Curriculum setup
+    curriculum = None
+    if args.curriculum:
+        sample_env = openai_gym.make(f"procgen-{args.env_id}-v0")
+        sample_env = GymV21CompatibilityV0(env=sample_env)
+        sample_env = ProcgenTaskWrapper(sample_env, args.env_id, seed=args.seed)
+
+        # Intialize Curriculum Method
+        if args.curriculum_method == "dr":
+            print("Using domain randomization.")
+            curriculum = DomainRandomization(sample_env.task_space)
+        else:
+            raise ValueError(f"Unknown curriculum method {args.curriculum_method}")
+        curriculum = make_multiprocessing_curriculum(curriculum)
+        del sample_env
+
+    # env setup
+    print("Creating env")
+    venv = DummyVecEnv(
     [
-        make_env(task_queue, update_queue, num_levels=0)
-        for i in range(64)
+        make_env(args.env_id,
+                args.seed + i,
+                curriculum=curriculum if args.curriculum else None,
+                num_levels=1 if args.curriculum else 0)
+        for i in range(args.num_envs)
     ]
-)
-venv = wrap_vecenv(venv)
-
-model = PPO(
-    "CnnPolicy",
-    venv,
-    verbose=1,
-    n_steps=256,
-    learning_rate=linear_schedule(0.0005),
-    gamma=0.999,
-    gae_lambda=0.95,
-    n_epochs=3,
-    clip_range_vf=0.2,
-    ent_coef=0.01,
-    batch_size=256 * 64,
-    tensorboard_log="runs/testing"
-)
-
-wandb_callback = WandbCallback(
-    model_save_path=f"models/{run.id}",
-    verbose=2,
-)
-plr_callback = CustomCallback(curriculum)
-callback = CallbackList([wandb_callback, plr_callback])
-model.learn(
-    25000000,
-    callback=callback,
-)
+    )
+    venv = wrap_vecenv(venv)
+    assert isinstance(venv.action_space, gym.spaces.discrete.Discrete), "only discrete action space is supported"
+
+    print("Creating agent")
+    agent = ProcgenAgent(
+        venv.single_observation_space.shape,
+        venv.single_action_space.n,
+        arch="large",
+        base_kwargs={'recurrent': False, 'hidden_size': 256}
+    ).to(device)
+
+    print("Creating model")
+    model = PPO(
+        "CnnPolicy",
+        venv,
+        verbose=1,
+        n_steps=256,
+        learning_rate=linear_schedule(0.0005),
+        gamma=0.999,
+        gae_lambda=0.95,
+        n_epochs=3,
+        clip_range_vf=0.2,
+        ent_coef=0.01,
+        batch_size=256 * 64,
```

Comment:
```suggestion
        batch_size=2048
```

batch_size is actually the minibatch size. We want 8 batches for 25*64 steps, so 2048 steps per minibatch